observatory-dags/observatory/dags/database/sql/aggregate_unpaywall.sql.jinja2

```diff
@@ -12,31 +12,182 @@
+preprint repositories (eg arxiv) both submitted, accepted and VoR versions of outputs. In a small number of cases
+preprint repositories register content as being journal articles (SSRN is the most common case). Green as
+defined here also explicitly includes those outputs that are also available via the publisher. For the set
+of content which is only freely available via a repository see `green_only`.
+#}
+  CASE
+    WHEN (SELECT COUNT(1) FROM UNNEST(oa_locations) AS location WHERE location.host_type IN ('repository')) > 0 THEN TRUE
+    ELSE FALSE
+  END
+    as green,
+
+{#
+### Green Open Access where the outputs is not available Gold (DOAJ or Hybrid) or Bronze:
+
+green_only is the subset of outputs available from repositories that are not also available free to read from
+the publisher. This category enables analyses of gold, bronze and green as mutually exclusive categories, e.g.
+in the generation of stacked bar charts that include gold_doaj, green, hybrid and bronze. This corresponds to
+general usage of the term "green" in some other literature.
+#}
+  CASE
+    WHEN (SELECT COUNT(1) FROM UNNEST(oa_locations) AS location WHERE location.host_type IN ('repository')) > 0 AND
+      NOT (journal_is_in_doaj OR best_oa_location.host_type = "publisher") THEN TRUE
+    ELSE FALSE
+  END
+    as green_only,
+
+{#
+### Green Open Access where the output is not available Gold (DOAJ or Hybrid) but may be Bronze:
+
+green_only_ignoring_bronze provides the set of articles that are green and not gold. That is it includes articles
+that are green and bronze, but not gold. This category enables analyses of gold and green as mutually
+exclusive categories, e.g. in the generation of stacked bar charts that include gold_doaj, green and hybrid.
+#}
+  CASE
+    WHEN (SELECT COUNT(1) FROM UNNEST(oa_locations) AS location WHERE location.host_type IN ('repository')) > 0 AND
+      NOT (journal_is_in_doaj OR (best_oa_location.host_type = "publisher" AND best_oa_location.license is not null)) THEN TRUE
+    ELSE FALSE
+  END
+    as green_only_ignoring_bronze,
+
+{#
+### Convenience category for analysing articles that have a license for the best OA location
+#}
+  CASE
+    WHEN (best_oa_location.license IS NOT NULL) THEN TRUE
+    ELSE FALSE
+  END
+    as has_license,
+
+{#
+### Convenience category for analysing articles that have a Creative Commons license for the best OA location
+#}
+  CASE
+    WHEN ((best_oa_location.license IS NOT NULL) AND (STARTS_WITH(best_oa_location.license, "cc"))) THEN TRUE
+    ELSE FALSE
+  END
+    as is_cclicensed,
+
+{#
+### Provides a list of repository URLs for future analysis by repository type
+#}
+  ARRAY((SELECT url FROM UNNEST(oa_locations) WHERE host_type = "repository")) as repository_locations,
   best_oa_location.url_for_landing_page,
   best_oa_location.url_for_pdf
+
+FROM `academic-observatory.our_research.unpaywall20201009`
```

Comment:
this line needs to be removed

Reply:
The difference in green_only is as expected 
(for my own reference as much as anyone else's:  the new query counts all articles in DOAJ journals as gold, also when Unpaywall labels them as only available from a repository)

The difference in green_only_ignoring_bronze is also indeed as expected 
(@cameronneylon the new query fixed an OR vs AND mistake, right? It now correctly excludes articles that are either gold_in_doaj *or* hybrid, which is correct))